src/enable_variable_ops/ngraph_var.h

```diff
@@ -76,17 +85,73 @@ class NGraphVar : public ResourceBase {
   }
 
   bool need_sync_ng_tensor() { return sync_ng_tensor_; }
-  void sync_ng_tensor(bool sync_ng_tensor) { sync_ng_tensor_ = sync_ng_tensor; }
+  void set_sync_ng_tensor(bool sync_ng_tensor) {
+    sync_ng_tensor_ = sync_ng_tensor;
+  }
+
+  // Copies the NG Tensor to TF Tensor for this variable
+  // Involves a copy from device to host
+  // Returns the number of copies made (0 or 1)
+  int copy_ng_to_tf() {
+    if (ng_tf_share_buffer_) {
+      return 0;
+    }
+    ngraph::Event event_sync_ng_tf_tensors("sync ng-tf-tensors D2H", "", "");
+    ReadNGTensor(ng_tensor_, &tf_tensor_);
+    event_sync_ng_tf_tensors.Stop();
+    ngraph::Event::write_trace(event_sync_ng_tf_tensors);
+    return 1;
+  }
 
-  // TODO(malikshr): Implement syncing utility functions here
-  Status copy_ng_to_tf();
-  Status copy_tf_to_ng();
+  // Copies the TF Tensor to NG Tensor for this variable
+  // Involves a copy from host to device
+  // Returns the number of copies made (0 or 1)
+  int copy_tf_to_ng() {
+    if (ng_tf_share_buffer_) {
+      return 0;
+    }
+    ngraph::Event event_sync_ng_tf_tensors("sync ng-tf-tensors H2D", "", "");
+    WriteNGTensor(ng_tensor_, &tf_tensor_);
+    event_sync_ng_tf_tensors.Stop();
+    ngraph::Event::write_trace(event_sync_ng_tf_tensors);
+    return 1;
+  }
+
+  // If the NGTensor is behind TF Tensor (ie if NGTensor is out-of-date)
+  // It updates ng_tensor by copy_tf_to_ng
+  // Returns the number of copies made (0 or 1)
+  int sync_ng_tensor() {
+    if (ng_tf_share_buffer_) {
```

Comment:
May be you can remove this check here since you already have it in **copy_tf_to_ng**